Remove `Clone` bound on `map_tabs` and `filter_map_tabs`

Ved-s opened the pull request:
No point of having a FnMut if it can't mutably reference stuff outside

Adanos020 reviewed (commented):
This seems like a good change. I noticed you haven't applied it to `{DockState,Surface,Tree}::{filter_tabs,retain_tabs}` and `{Surface,Tree}::map_tabs`, is there a particular reason for this?

Also, since this is an API change, I'll move it to `release-0.13`.

Ved-s commented:
> I noticed you haven't applied it to `{DockState,Surface,Tree}::{filter_tabs,retain_tabs}` and `{Surface,Tree}::map_tabs`, is there a particular reason for this?

I guess i just removed it from the stuff i used and didn't check for other instances of it.
Fixed

Adanos020 reviewed (approved):
Sorry for the massive delay, my irl stuff has been insane in the last 2 months.

The changes look good, merging them soon after I've updated the changelog. Thank you!
